Project Configuration section overview (#808)
o3de/o3de.org: content/docs/user-guide/project-config/_index.md

To create and manage projects, you can use either the GUI-based **Project Manager** tool or the O3DE command line interface (CLI). The topics in this section provide details on creating O3DE projects using either of these options.

| Topic | Description |
| - | - |
| [Understanding project.json](#understanding-project-json) | A detailed look at the fields in the project manifest file, `project.json`. |
| [Adding and Removing Gems](add-remove-gems/) | Learn how to add and remove Gems in your project. | 
| [Registering Gems](register-gems/) | Learn how to register Gems that are provided outside of the O3DE source. | 
| [Troubleshooting](troubleshooting/) | A troubleshooting guide for project configuration. |

## Understanding `project.json` {#understanding-project-json}

Each project contains a project manifest file named `project.json`. This file is located in the project root directory, and stores important configuration properties for each project. It is created for you at the time of project creation.

> To make changes to any of the properties in this file, you can either edit it manually using an editor, or specify individual properties to edit using the [O3DE CLI](./cli-reference) `edit-project-properties` command.

Reviewer: Clarifies that you are editing the text of the file. 
```suggestion
To make changes to any of the properties in this file, you can either edit it manually using a text editor, or specify individual properties to edit using the [O3DE CLI](./cli-reference) `edit-project-properties` command.
```

The following table describes each of the properties in `project.json`. The name given to the project when the project was created is used as part of the default value for many of the properties, and is indicated by "<PROJECT_NAME>".

| Property | Description | Default |
| --- | --- | --- |
| **project_name** | The name of the project. Used by the O3DE CLI `--project-name` parameter to identify the project. | "<PROJECT_NAME>" |

> | **origin** | A URL for the project, such as the repository URL or project website. TBD: displayed where? | "The primary repo for <PROJECT_NAME> goes here: i.e. http://www.mydomain.com"

Reviewer: Highlighting as a reminder to update this.

---
linktitle: Project Configuration
title: Project Configuration
description: Learn the basics of project configuration in Open 3D Engine, and get the details on the Project Manager and O3DE CLI tools.
weight: 100
---

Open 3D Engine (O3DE) projects define and configure the set of code and assets that make up your game or other application. Some code and assets are provided by [Gems](/docs/user-guide/gems) that you enable for your project. Projects are contained within a project directory, sometimes referred to as the "project root".

To create and manage projects, you can use either the GUI-based **Project Manager** tool or the O3DE command line interface (CLI). The topics in this section provide details on creating O3DE projects using either of these options.

| Topic | Description |
| - | - |
| [Understanding project.json](#understanding-project-json) | A detailed look at the fields in the project manifest, an important project configuration file. |
| [Adding and Removing Gems](add-remove-gems/) | Learn how to add and remove Gems in your project. | 
| [Registering Gems](register-gems/) | Learn how to register external Gems from sources outside of O3DE. |

> | [O3DE CLI Reference](cli-reference/) | A command line interface (CLI) reference for the `o3de` Python script.

Author: Actually, the O3DE CLI is for more than just projects. I'll start with your reworded suggestion and go from there, updating the cli-reference topic while I'm at it.

To build projects, you can use either Project Manager or **CMake**. For more information, refer to the [introductory tutorials](/docs/welcome-guide/create/) in the Get Started Guide, and the [Configure and Build](/docs/user-guide/build/configure-and-build) topic in the Build section of this User Guide.

## Understanding `project.json` {#understanding-project-json}

Each project's root directory contains a project manifest file named `project.json`. This file stores important project configuration properties. O3DE creates this file for you during project creation.

To change any of the properties in this file, you can either edit the file manually using a text editor, or edit individual properties using the `edit-project-properties` [O3DE CLI](./cli-reference) command. Additionally, you can edit the `display_name` property in Project Manager.

The following table describes each of the properties in `project.json`. The default value for many of the properties contains the project name given when you created the project. We indicate this project name as "<PROJECT_NAME>".

| Property | Description | Default |
| --- | --- | --- |

> | **project_name** | The name of the project. The `--project-name` O3DE CLI  parameter uses this name to identify the project. | "<PROJECT_NAME>" |

Reviewer 2: ```suggestion
| **project_name** | The name of the project. The `--project-name` O3DE CLI parameter uses this name to identify the project. | "<PROJECT_NAME>" |
```
Looks like I accidentally introduced an extra space in my suggested edit.

> | **origin** | A URL for the project, such as the repository URL or project website. This field is intended to be available for display and use by Project Manager. | "The primary repo for <PROJECT_NAME> goes here: i.e. http://www.mydomain.com" |

Author: I actually meant "for Project Manager to display and use". I'll update.

| **license** | The license your project uses, and any copyright information you want to include. Consider providing the URL for the license. This field is intended to be available for display and use by Project Manager. | "What license <PROJECT_NAME> uses goes here: i.e. https://opensource.org/licenses/MIT" |
| **display_name** | The display name of the project in Project Manager. | "<PROJECT_NAME>" |

> | **summary** | A short description of your project. This field is intended to be available for display and use by Project Manager. | "A short description of <PROJECT_NAME>." |

Reviewer 2: ```suggestion
| **summary** | A short description of the project. This field is intended for display and for Project Manager to use. | "A short description of <PROJECT_NAME>." |
```
Changed "your project" to "the project" for consistency with most other descriptions in the table. (Using "your" instead of "the" is totally fine; just be consistent.)

> | **canonical_tags** | A standard field in O3DE manifests used to identify what type of O3DE object this is. Examples: "Gem", "Project". Projects should use the "Project" tag. | [ "Project" ] |

Reviewer 2: ```suggestion
| **canonical_tags** | A standard field in O3DE manifests used to identify O3DE object types. Examples: "Gem", "Project". Projects should use the "Project" tag. | [ "Project" ] |
```
Or, "to identify an O3DE object's type."

Author: I like "O3DE object's type" here.

> | **user_tags** | Any keyword tags that you want to associate with your project. Examples: "Physics", "Assets", "AWS". You can see examples of the standard set of tags used by Gems included in the Project Manager **Gem Catalog**. These tags are used by Project Manager for documentation, search, and filtering. | [ "<PROJECT_NAME>" ] |

Reviewer 2: ```suggestion
| **user_tags** | Any keyword tags that you want to associate with the project. Examples: "Physics", "Assets", "AWS". For examples of the standard set of tags that Gems use, refer to the Project Manager **Gem Catalog**. Project Manager uses these tags for documentation, search, and filtering. | [ "<PROJECT_NAME>" ] |
```
Edits for cross-reference link wording ("see" > "refer to"), active voice

> | **icon_path** | The path and filename of the project icon. This icon is used in **Project Manager**. It must be named `preview.png` and located in the project root directory. | "preview.png" <br> DO NOT CHANGE! |

Reviewer 2: ```suggestion
| **icon_path** | The path and filename of the project icon. Project Manager uses this icon. The file must be located in the project root directory, and the filename must be `preview.png`. <br><br> WARNING: Do not change the path or filename. | "preview.png" |
```
* Dropped the **bold** formatting from "Project Manager," as the tool is mentioned frequently prior to this description. (Use bold only on first occurrence.)
* Changed the ambiguous "it" ("...Project Manager. It....") to "the file."
* Edit for active voice ("used in Project Manager" > "Project Manager uses")
* Re-ordered the second sentence to bring the location/path directive to the front of the sentence. This is more consistent with the property name ("icon_path") and the earlier wording order ("path and filename").
* Reworded the warning from "DO NOT CHANGE!" to a more formal "WARNING" note/alert.
  * Consider moving this into the description. IMO, it gets lost in the "Default" column, visually blurring with the all-caps "PROJECT_NAME".

> | **engine** | The name of the engine registered to the project. Engines get registered in your O3DE manifest: `<USER>/.o3de/o3de_manifest.json`. | "o3de" |

Reviewer 2: ```suggestion
| **engine** | The name of the engine registered to the project. Register engines in your O3DE manifest, located at `<USER>/.o3de/o3de_manifest.json`. | "o3de" |
```

> | **restricted_name** | (Optional) The name of a restricted folder to associate with this project. | "projects" |

Reviewer 2: ```suggestion
| **restricted_name** | (Optional) The name of a restricted folder to associate with the project. | "projects" |
```

> | **external_subdirectories** | (Optional) The path to one or more directories to include in the project build. Any directory with a `CMakeLists.txt` file can be used here. The paths to Gems that are registered to the project using the O3DE CLI `register --external-subdirectory-project-path` command are added here. | |

Reviewer 2: ```suggestion
| **external_subdirectories** | (Optional) The path to one or more directories to include in the project build. You can use any directory with a `CMakeLists.txt` file. When you register Gems to the project using the `register --external-subdirectory-project-path` O3DE CLI command, O3DE adds their paths here. | |
```
Edits for active voice
